predictor.h

```diff
@@ -14,8 +14,47 @@
   concerned about the memory used by the simulated branch predictor.
 */
 
+#include <math.h>
 
+# define ST 0
+# define WT 1
+# define WNT 2
+# define SNT 3
 
+/*
+Budget sizes
+
+8K   + 64  -> k=6,  s=7
+16K  + 128 -> k=7,  s=7
+32K  + 256 -> k=8,  s=7
+64K  + 512 -> k=9,  s=7
+128K + 1K  -> k=10, s=7
+1M   + 4K  -> k=12, s=8
+*/
+
+//# define k 12 // number of bits for hashing branches
+//# define s 8 // 4096=2^12.  12 bits for history
+
+
+/*
+Stores global history table
+
+Maximum size
+k=12 -> 4096 = 2^12
+s=8  -> 256  = 2^12
+ */
+//extern int pattern_table[4096][256];
+
+/*
+Histories for each branch
+
+Maximum size
+k=12 -> 4096 = 2^12
+*/
+//extern int histories[4096];
+
+// helper function that updates the global table;
+void step(int i, int j, bool outcome);
```

Comment:
Might want to remove this and the corresponding test.  The header shouldn't be changed I don't think.

Not sure how to address their comment about adding the tables to the predictor.h 
There are going to be different tables for each predictor.